Closes #82 vignette update (#83)
pharmaverse/admiralpeds: vignettes/advs.Rmd

### Reference Files for Parameters by Age

As growth rates and patterns by age and sex, reference values differ by these

> two characteristics. 

Reviewer: @Fanny-Gautier @zdz2101 does this sentence read well to you? it doesn't make sense to me. i found the old wording clearer but might just be me

Author: I am not a native English speaker, I had first to translate it to ensure I properly understood it... We can revert it or change it again if you prefer. I am totally fine with this option ;-)

Reviewer: Revert it please. I think its missing a word or 2 for it to read properly.

Author: I think that the correct sentence should have been "As growth rates and patterns **differ** by age and sex, reference values differ by these two characteristics. "

What do you think instead of "As growth rates and patterns differ between sex, different reference values are used accordingly and for different age groups.", as the part "and for different age groups." is also a bit weird.

For the various measures, we use the WHO reference data for children <2 years 
of age (\<730.5 days) and CDC growth charts for children >=2 years of age 
(\>=730.5 days).
This is in accord with the [CDC
recommendation](https://www.cdc.gov/growthcharts/who_charts.htm). The reference 
data for both charts include the LMS parameters, in which L is the Box-Cox 
transformation for normality, M is the median, and S is the coefficient of 

> variation (S). 

Reviewer: ```suggestion
variation. 
```
doesn't need the brackets

)
```

For BMI in the CDC metadata, a dispersion parameter (Sigma) is used to calculate
BMI percentiles and z-scores above the sex- and age-specific 95th percentile (P95).  

The above example only shows the parameter BMI, but this follows other parameters
like weight and height. We can combine WHO and CDC metadata for height as a
WHO adjustment is unnecessary.

> Note that the head circumference parameter metadata comes only from the WHO standards,

Reviewer: ```suggestion
Note that the head circumference parameter metadata comes only from the WHO reference data,
```

Reviewer: Given David's other comments/updates

as CDC provides no equivalent for children \> 2 years of age.

### Reference Files for Weight by Length/Height

> WHO provides additional standards for weight-for-length (recumbent length), 

Reviewer: ```suggestion
WHO provides additional reference data for weight-for-length (recumbent length),
```

> instead of age. Again, we combined the metadata for sex to create a single 

Reviewer: ```suggestion
instead of age. Again, we combined the metadata for sex to create a single
```
remove any blank spaces at the end of lines

reference file. 

There are also weight-by-height files available, but upon advice from a retired 
CDC expert in this area, we chose only to include the weight-for-length files, 
as once children can stand, BMI by age is a more appropriate measure to use.
However, in our template, we give the metadata values variable a generic name,
`HEIGHT_LENGTH`, so that any user could equally choose to pass weight-for-height

> files if preferred. 

Reviewer: ```suggestion
files if preferred.
```

## Initial ADVS Set-up {#advs_start}

> The following steps are to read in the source data, merge ADSL variables, 

Reviewer: ```suggestion
The following steps are to read in the source data, merge ADSL variables,
```

> and derive the usual ADVS analysis variable. The

Reviewer: ```suggestion
and derive the usual ADVS analysis variables. The
```

## Derive Additional Variables for Anthropometric indicators {#derive_vars}

To compare against the reference files, we need to know each child's age and
length/height at the time of each vital signs assessment. 
So, these need to be first derived as variables in your ADVS.

Remember to ensure that the unit of these matches that of the reference
files metadata you are comparing against, i.e. "days" for age and "cm"
for length/height in our examples here.

Note that some growth reference files we use here have different units and intervals. 
For example when you look at the "by Age" charts, WHO provides this by day,
but the CDC gives this every month (25.5, 26.5, etc.). So, converting all ages 
to days leaves the challenge of joining when the patient's current age falls 

> between the monthly CDC reference data. In our package functions, we used the

Reviewer 2: I'm not sure what's the best way to reword exactly what we did or just get rid of this paragraph in its entirety

> between the monthly CDC reference data. In our package functions, we used the

Reviewer: i would remove all this section (lines 313-321). i think its a carry over from before we added the interpolation piece above which now covers this

## Derive Additional Parameters for Anthropometric indicators {#derive_params}

Now, we get to the most important section, which shows how to create new records
for each derived pediatrics parameter.
Only specific examples are included here, but you'll find more parameters in the

> example template script referenced at the bottom of this vignette. 

Reviewer: ```suggestion
example template script referenced at the bottom of this vignette.
```

  )
```

If only a z-score or percentile were needed, you leave out one of the `set_values_to_`
arguments.

Once again, it is essential to remember the importance of the metadata being
supplied to the above function via the `meta_criteria` argument.

> Our default case from this package uses WHO standards for children \< 2 years of

Reviewer: ```suggestion
Our default case from this package uses WHO reference data for children \< 2 years of
```

age and CDC reference data for children \>= 2 years of age. As the user, you can 

> use other (e.g., IOTF) reference data files created with the LMS method.

Reviewer: ```suggestion
use other (e.g., International Obesity Task Force) reference data files created with the LMS method.
```

Reviewer: can also then remove this from wordlist file

```

For height parameters by age, it should be noted that the WHO growth chart
we use for children \<2 years of age refers to body length, whereas
the CDC growth chart for children \>=2 years of age refers to height.
This is explained thoroughly in the next section.

### Derived Weight by Length/Height

The following code is for Weight by Length/Height z-score and percentile for
children \< 2 years of age, which assumes the "HEIGHT" parameter is always

> collected as recumbent length. 

Reviewer: ```suggestion
collected as recumbent length.
```

  )
```

> We consulted with a pediatrician with extensive experience working with length 

Reviewer: ```suggestion
We consulted with a pediatrician with extensive experience working with length
```

> and height data, who advised that analysts rarely know whether standing height 

Reviewer: ```suggestion
and height data, who advised that analysts rarely know whether standing height
```

> or recumbent length was measured. It depends mainly on the age at which the 

Reviewer: ```suggestion
or recumbent length was measured. It depends mainly on the age at which the
```
